### PR TITLE
Elevenlabs detector

### PR DESCRIPTION
**Pull request by dylanTruffle**
### Description:
ElevenLabs is a leading text-to-speech platform. Most notably powers Rabbit R1, where the API key was recently compromised. This detector will hopefully help catch such instances in the future before they leak out.

https://elevenlabs.io/
https://rabbitu.de/articles/security-disclosure-1

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

**Comment by ahrav** [comment]
looks like this might be a dupe? https://github.com/trufflesecurity/trufflehog/pull/3021

**Comment by dxa4481** [comment]
Woops... yup @ahrav I didn't notice Ahmed opened a PR already. There are 2 versions of the key type, my PR has both, I think his just has the latest.

File: pkg/detectors/elevenlabs/v1/elevenlabs.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+var (
+	defaultClient = common.SaneHttpClient()
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyPat = regexp.MustCompile(`(?i)(?:elevenlabs|xi-api-key|el|token|key)[^\.].{0,40}[ =:'"]+([a-f0-9]{32})\b`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"elevenlabs","xi-api-key"}
```
Comment (0x1) [inline pkg/detectors/elevenlabs/v1/elevenlabs.go:42]: i would add `xi_api_key` also since that's a common variable the key is set to

**Review by 0x1 — APPROVED** [review]

**Review by Hamidsalri — APPROVED** [review]

File: pkg/detectors/elevenlabs/v1/elevenlabs.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+	return
+}
+
+func verifyMatch(ctx context.Context, client *http.Client, token string) (bool, *UserRes, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "https://api.elevenlabs.io/v1/user", nil)
+	req.Header.Set("xi-api-key", token)
+	if err != nil {
+		return false, nil, nil
+	}
```
Comment (mcastorina) [inline pkg/detectors/elevenlabs/v1/elevenlabs.go:87]: Should the error be returned here?

File: pkg/detectors/elevenlabs/v2/elevenlabs.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+	return
+}
+
+func verifyMatch(ctx context.Context, client *http.Client, token string) (bool, *UserRes, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "https://api.elevenlabs.io/v1/user", nil)
+	req.Header.Set("xi-api-key", token)
+	if err != nil {
+		return false, nil, nil
+	}
```
Comment (mcastorina) [inline pkg/detectors/elevenlabs/v2/elevenlabs.go:87]: Same here

File: pkg/detectors/elevenlabs/v1/elevenlabs.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+	for _, match := range keyPat.FindAllStringSubmatch(dataStr, -1) {
+		uniqueMatches[match[1]] = struct{}{}
+	}
+
+	for match := range uniqueMatches {
+		s1 := detectors.Result{
+			DetectorType: detectorspb.DetectorType_ElevenLabs,
+			Raw:          []byte(match),
+			ExtraData:    map[string]string{"version": "1"},
+		}
```
Comment (mcastorina) [inline pkg/detectors/elevenlabs/v1/elevenlabs.go:59]: Do we need `RawV2` here too?

**Review by mcastorina — APPROVED** [review]
overall lgtm, just had a few questions

File: pkg/detectors/elevenlabs/v1/elevenlabs.go (hunk 2 above)
Comment (0x1) [reply pkg/detectors/elevenlabs/v1/elevenlabs.go:87]: indeed

**Pull request merged**